app/code/Magento/Catalog/Model/Product/Type/Price.php

```diff
@@ -480,23 +489,23 @@ public function getTierPriceCount($product)
     /**
      * Get formatted by currency tier price
      *
-     * @param   float $qty
-     * @param   Product $product
+     * @param float $qty
+     * @param Product $product
      *
-     * @return  array|float
+     * @return array|float
      * @since 102.0.6
      */
     public function getFormattedTierPrice($qty, $product)
     {
         $price = $product->getTierPrice($qty);
         if (is_array($price)) {
             foreach (array_keys($price) as $index) {
-                $price[$index]['formatted_price'] = $this->priceCurrency->convertAndFormat(
+                $price[$index]['formatted_price'] = $this->amount->convertAndFormatCurrency(
                     $price[$index]['website_price']
                 );
             }
         } else {
```

Comment:
As done before to improve readability, the _return early_ pattern can also be applied here, returning `$price` at the end of the loop and avoiding the `else` statement.